lib/sqlalchemy/sql/schema.py

```diff
@@ -4549,7 +4576,10 @@ def _set_parent(self, parent: SchemaEventTarget, **kw: Any) -> None:
 
         for c in self._columns:
             c.primary_key = True
-            if c._user_defined_nullable is NULL_UNSPECIFIED:
+            if (
+                not isinstance(c, Period)
+                and c._user_defined_nullable is NULL_UNSPECIFIED
+            ):
                 c.nullable = False
```

Comment:
**mike bayer** (zzzeek) wrote:

or this.....   the Period concept is this very new and special thing that's kind of an accessory to the general concept of tables, columns and constraints.  it should stay on the outside of things as much as possible to prevent it from becoming burdensome in places like this.

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/3803